src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -227,7 +230,10 @@ public override TValue VisitFlowCaptureReference (IFlowCaptureReferenceOperation
 		// is like a local reference.
 		public override TValue VisitFlowCapture (IFlowCaptureOperation operation, LocalDataFlowState<TValue, TValueLattice> state)
 		{
-			TValue value = Visit (operation.Value, state);
+			// If the captured value is a property reference, we can't easily tell inside of
+			// VisitPropertyReference whether it is accessed for reads or writes.
+			// https://github.com/dotnet/roslyn/issues/25057
+			// Avoid visiting the captured value unless it is an RValue.
```

Comment:
Just curious - won't this cause similar problems for properties like we have with not seeing the "instance" operation for field access? Basically a property variant of https://github.com/dotnet/linker/issues/2832